core/kernel_module.go

```diff
@@ -139,6 +140,22 @@ func (m *kernelModule) generateKbuildArgs(ctx blueprint.ModuleContext) map[strin
 		kbuildOptions = "--kbuild-options " + strings.Join(m.build().Kbuild_options, " ")
 	}
 
+	hostToolchain := ""
+	if props.GetBool("host_toolchain_clang") {
+		hostToolchain = "--hostcc clang"
```

Comment:
Here, "clang" should be HOST_CLANG_PREFIX + CLANG_CC_BINARY. Ideally we want to get this from toolchain

Reply:
I've changed this part to get tool from toolchain